__tests__/app.test.js

```diff
@@ -31,9 +33,589 @@ describe("NCNews-Server Unit Tests", () => {
+            });
+
+        });
+
+        test(`[ 400 ] Responds with an error when passed an object without the required keys`, () => {
+
+            const votesObject = {
+                wat: "k",
+                k: "k"
+            };
+
+            return request(app).post('/api/articles/1/comments').send(votesObject).expect(400).then((response) => {
+                expect(response.body).toEqual({ error: "Error 400! BAD REQUEST" });
+            });
+
+        });
+
+        test(`[ 400 ] Responds with an error when passed an invalid vote of the wrong type`, () => {
+
+            const votesObject = {
+                inc_votes: `dksfsdjlfds`
+            };
+
+            return request(app).post('/api/articles/1/comments').send(votesObject).expect(400).then((response) => {
+                expect(response.body).toEqual({ error: "Error 400! BAD REQUEST" });
+            });
+
+
+        });
+
+        test(`[ 400 ] Responds with an error when passed an SQL injection in the votes property`, () => {
+
+            const votesObject = {
+                inc_votes: `1; DROP TABLE articles;`
+            };
+
+            return request(app).post('/api/articles/1/comments').send(votesObject).expect(400).then((response) => {
+                expect(response.body).toEqual({ error: "Error 400! BAD REQUEST" });
+            });
+
+
+        });
+
+        
+        test(`[ 400 ] Responds with an error when passed an SQL injection in URL parameters, test 2`, () => {
+
+            return request(app).post(`/api/articles/1; DROP TABLE articles;/comments`).send(defaultVotesObject).expect(400).then((response) => {
+                expect(response.body).toEqual({ error: "Error 400! BAD REQUEST" });
+            });
+
+        });
+
+    }); 
+
+    // Task 9 - GET /api/users
+    describe(`GET /api/users/`, () => {
+
+        test(`[ 200 ] Responds with an array of objects, where the objects have the following 3 properties: username, name, avatar_url`, () => {
+
+            return request(app).get(`/api/users`).expect(200).then((response) => {
+                const usersArray = response.body.users;
+                expect(usersArray).toHaveLength(4);
+
+                usersArray.forEach((userObject) => {
+                    expect(userObject).toEqual(
+                        expect.objectContaining({
+                            username: expect.any(String),
+                            name: expect.any(String),
+                            avatar_url: expect.any(String)
+                    }));
+                });
+            });
+        });
+           
+    });
+
+    // Task 12 - DELETE /api/comments/:comment_id
+
+    describe(`DELETE /api/comments/:comment_id`, () => {
+
+        test(`[ 204 ] Responds with status 204 and no content`, () => {
+
+            return request(app).delete('/api/comments/1').send().expect(204).then((response) => {
+                
+                expect(response.body).toEqual({});
```

Comment:
I think just testing for a 204 is fine, the 204 response should not allow a response body anyway :)